interfaces/api/service/IMapsManager.h

```diff
@@ -114,6 +134,51 @@ class XPCF_CLIENTUUID("4863f2d1-023c-4095-8842-e6083e6ec54a") XPCF_SERVERUUID("2
         const std::string & mapUUID,
         SRef<SolAR::datastructure::PointCloud> & pointCloud) const = 0;
 
+    /// @brief Register a new Map Processing service to the map manager
+    /// @param[in] processingType the type of process managed by the service
+    /// @param[in] serviceURL URL of the Map Processing service
+    /// @return FrameworkReturnCode::_SUCCESS if the Map Processing service is registered, else FrameworkReturnCode::_ERROR_
+    virtual FrameworkReturnCode registerMapProcessingService(const MapProcessingType processingType,
+                                                             const std::string & serviceURL) = 0;
+
+    /// @brief Unregister a Map Processing service from the map manager
+    /// @param[in] processingType the type of process managed by the service
+    /// @param[in] serviceURL URL of the Map Processing service
+    /// @return FrameworkReturnCode::_SUCCESS if the Map Processing service is unregistered, else FrameworkReturnCode::_ERROR_
+    virtual FrameworkReturnCode unregisterMapProcessingService(const MapProcessingType processingType,
+                                                               const std::string & serviceURL) = 0;
+
+    /// @brief Request for a map processing giving the type of process to apply (asynchronous)
+    /// @param[in] mapUUID the UUID of the map to process
+    /// @param[in] processingType the type of process to apply on the map
+    /// @return FrameworkReturnCode::_SUCCESS if processing is able to proceed
+    ///         FrameworkReturnCode::_NO_SERVICE_AVAILABLE if a necessary service is not available
+    ///         else FrameworkReturnCode::_ERROR_
+    virtual FrameworkReturnCode requestMapProcessing(const std::string & mapUUID,
+                                                     const MapProcessingType processingType) = 0;
+
+    /// @brief Get status and progress percentage concerning a map processing in progress
+    ///        If status = COMPLETED then give the map UUID of the new resulting map
+    /// @param[in] mapUUID the UUID of the map being processed
+    /// @param[out] status the current map processing status
+    /// @param[out] progress the current progress percentage (valid value should be between 0 and 1)
+    /// @param[out] resultingMapUUID the map UUID of the new created map (processing result)
+    /// @return FrameworkReturnCode::_SUCCESS if the status and progress are available, else FrameworkReturnCode::_ERROR_
+    virtual FrameworkReturnCode getMapProcessingStatus(const std::string & mapUUID,
+                                                       MapProcessingStatus status,
```

Comment:
Same here, `status` should be passed by reference

Reply:
done